webhook-notifications_5def901604286364bc92af37.md

```diff
@@ -0,0 +1,283 @@
+Semaphore has integrated Webhook based notifications that are delivered on
+the success or failure of a pipeline.
+
+- [Setting up Webhook notification for a project](#setting-up-webhook-notifications-for-a-project)
+- [Setting up Webhook notification for multiple projects](#setting-up-webhook-notifications-for-multiple-projects)
+- [Filtering by project, branch and pipeline names](#filtering-by-project-branch-and-pipeline-names)
+- [Advanced notification setup](#advanced-notification-setup)
+  - [Filtering by pipeline result](#filtering-by-pipeline-result)
+- [Modifying notification settings](#modifying-notification-settings)
+- [Setting up, editing and deleting Webhook notifications through the UI](#setting-up-editing-and-deleting-webhook-notifications-through-the-ui)
+- [Notification payload](#notification-payload)
+- [See also](#see-also)
+
+## Setting up Webhook notifications for a project
+
+Use your Endpoint URL to set up a notification on Semaphore,
+with the following command:
+
+``` bash
+$ sem create notification [name] \
+    --projects [project_name] \
+    --webhook-endpoint [webhook-endpoint]
+```
+
+For example, if you have a project called `web` and you want to get a Webhook
+notification on every finished pipeline on the `master` branch, use the
+following command:
+
+``` bash
+$ sem create notification master-pipelines \
+    --projects web \
+    --branches master \
+    --webhook-endpoint [webhook-endpoint]
+```
+
+## Setting up Webhook notifications for multiple projects
+
+When creating the notification, you can specify multiple projects as source,
+of your notifications.
+
+For example, if your team manages three projects named `web`, `cli` and `api`
+and you want to get notified for every finished pipeline on the master branch,
+use the following command:
+
+``` bash
+$ sem create notifications teamA-notifications \
+    --projects "web,cli,api" \
+    --branches "master" \
+    --webhook-endpoint [webhook-endpoint]
+```
+
+## Filtering by project, branch and pipeline names
+
+When creating the notification, you can specify a filter for project, branch and
+pipeline names.
+
+For example to send notifications for the `master` and `staging` branches use
+the following:
+
+``` bash
+$ sem create notifications example \
+    --branches "master,staging" \
+    --webhook-endpoint [webhook-endpoint] \
+```
+
+The branch filter can be a direct match like in the previous example, or a
+regular expression match. For example, to get notified about for `master` and
+every branch that matches `hotfix/-`, use the following:
+
+``` bash
+$ sem create notifications example \
+    --branches "master,/hotfix\/.*/" \
+    --webhook-endpoint [webhook-endpoint] \
+```
+
+Regex matches must be wrapped in forward slashes (example: `/.*/`). Specifying a
+branch name without slashes (example: `.*`) would execute a direct equality
+match.
```

Comment:
A filter can either be a regex match or a direct match. Specifying a filter with forward slashes (e.g. `/hotfix-.*/`) defines a regex match, while a filter without forward slashes (e.g. `master`) defines a direct match.